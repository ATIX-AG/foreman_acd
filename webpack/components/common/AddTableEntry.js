import React from 'react';
import PropTypes from 'prop-types';
import { Icon, Button } from 'patternfly-react';

const AddTableEntry = ({ hidden, disabled, onAddTableEntry }) => {
  if (hidden) {
    return null;
  }

  return (
    <Button bsStyle="default" disabled={disabled} onClick={onAddTableEntry}>
      <Icon type="fa" name="plus" />
    </Button>
  );
};

AddTableEntry.propTypes = {
  hidden: PropTypes.bool.isRequired,
  disabled: PropTypes.bool.isRequired,
  onAddTableEntry: PropTypes.func.isRequired,
};

export default AddTableEntry;
