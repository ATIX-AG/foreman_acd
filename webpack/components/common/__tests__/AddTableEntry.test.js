import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import AddTableEntry from '../AddTableEntry';

const noop = () => {};

const fixtures = {
  'should render add parameter': {
    hidden: false,
    disabled: false,
    onAddTableEntry: noop,
  },
  'should render hidden button': {
    hidden: true,
    disabled: false,
    onAddTableEntry: noop,
  },
  'should render disabled button': {
    hidden: false,
    disabled: true,
    onAddTableEntry: noop,
  },
};

describe('AddTableEntry', () =>
  testComponentSnapshotsWithFixtures(AddTableEntry, fixtures));
