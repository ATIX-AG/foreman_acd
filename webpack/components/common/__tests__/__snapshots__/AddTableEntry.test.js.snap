// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTableEntry should render add parameter 1`] = `
<Button
  active={false}
  block={false}
  bsClass="btn"
  bsStyle="default"
  disabled={false}
  onClick={[Function]}
>
  <Icon
    name="plus"
    type="fa"
  />
</Button>
`;

exports[`AddTableEntry should render disabled button 1`] = `
<Button
  active={false}
  block={false}
  bsClass="btn"
  bsStyle="default"
  disabled={true}
  onClick={[Function]}
>
  <Icon
    name="plus"
    type="fa"
  />
</Button>
`;

exports[`AddTableEntry should render hidden button 1`] = `""`;
